koku/api/user_access/view.py

```diff
@@ -28,121 +28,88 @@ class UIFeatureAccess:
         1. to keep UI and API in sync about RBAC permissions
         2. to provide the UI (both our UI and the platform's chroming) with a simple boolean response regarding whether
             a user has access to specific features of the UI.
-
-    Class attributes:
-
-        access_keys (list) - a list of access keys to check. must be implemented by sub-classes.
-                             multiple keys will be ORed; i.e. if a user has access to anything in the list,
-                             this API returns True
     """
 
-    def __init__(self, access):
+    PRERELEASE = ["ibm"]
+
+    def __init__(self, access, admin_user, access_key_list=None):
         """Class Constructor.
 
         Args:
             access (dict) - an RBAC dict; see: koku.koku.middleware.IdentityHeaderMiddleware
 
         """
+        if not access_key_list:
+            self.access_key_list = self.access_key_mapping.keys()
         self.access_dict = access if access else {}
+        self.admin_user = admin_user
+        self.access_key_list = []
+
+    def set_access_key_list(self, query_param):
+        """Checks the access key list."""
+        if query_param:
+            if query_param in self.access_key_mapping:
+                self.access_key_list = [query_param]
+        else:
+            self.access_key_list = self.access_key_mapping.keys()
+        return bool(self.access_key_list)
+
+    @property
+    def access_key_mapping(self):
+        mapping = {}
+        mapping["aws"] = ["aws.account", "aws.organizational_unit"]
+        mapping["ocp"] = ["openshift.cluster", "openshift.node", "openshift.project"]
+        mapping["azure"] = ["azure.subscription_guid"]
+        mapping["gcp"] = ["gcp.account", "gcp.project"]
+        mapping["oci"] = ["oci.payer_tenant_id"]
+        mapping["ibm"] = ["ibm.account"]
+        mapping["cost_model"] = ["cost_model"]
+        mapping["settings"] = ["settings"]
+        mapping["any"] = mapping["aws"] + mapping["azure"] + mapping["gcp"] + mapping["oci"] + mapping["ocp"]
+        return mapping
 
     def _get_access_value(self, key1, key2, default=None):
         """Return the access value from the inner dict."""
         return self.access_dict.get(key1, {}).get(key2, default)
 
-    def access(self, admin_user, pre_release_feature):
+    def _check_access(self, pre_release_feature, access_key):
         """Access property returns whether the user has the requested access.
 
         Return:
-            (bool)
+            (access, read_only)
+            acesss = bool or None
+            read_only = bool or none
```

Comment:
These returns are a bit confusing. Why would `read_only` ever be `None`? Also, `access` never returns `None`.

Reply:
In this case, `no access` is the same as `False`.

Reply:
Actually, now I'm confused, why does `read-only: True` equate to write access?

Reply:
Sorry, I have been thinking about his since I commented it and chatting with Dan about it. I think the real requirement here is to understand if the user has write access. So, I changed the PR to `write` which will make this boolean make way more sense.